packages/common/src/scorekeeper/jobs/Job.ts

```diff
@@ -0,0 +1,77 @@
+import { startJob } from "./cron/StartCronJobs";
+import logger from "../../logger";
+import { jobStatusEmitter } from "../../Events";
+import { JobStatus, JobConfig, JobRunnerMetadata } from "./types";
+
+export class Job {
+  protected status: JobStatus;
+  protected jobConfig: JobConfig;
+  protected jobRunnerMetadata: JobRunnerMetadata;
+
+  // TODO: remove this dependency injection
+  private startJobFunction: (
+    metadata: JobRunnerMetadata,
+    jobConfig: JobConfig,
+  ) => Promise<void>;
+
+  // TODO: remove events and use db to handle the state
+  // then we can decouple scorekeeper and gateway
+  static events: string[] = [
+    "jobStarted",
+    "jobRunning",
+    "jobFinished",
+    "jobErrored",
+    "jobProgress",
+  ];
+
+  constructor(jobConfig: JobConfig, jobRunnerMetadata: JobRunnerMetadata) {
+    this.status = {
+      name: jobConfig.name,
+      updated: Date.now(),
+      status: "Not Running",
```

Comment:
this should be of type StatusName, right ? can we avoid to have the string `Not Running` hardcoded ?

Reply:
I partially answered in the previous comment. Please let me know if you want me to include this in the current PR.
For the renaming issue, there is a dependency at least on `scorekeeper-status-ui`.
For the hardcoding issue, it's easier, but the plan was to tackle it together.